src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3701,6 +3710,62 @@ theorem udiv_twoPow_eq_of_lt {w : Nat} {x : BitVec w} {k : Nat} (hk : k < w) : x
   have : 2^k < 2^w := Nat.pow_lt_pow_of_lt (by decide) hk
   simp [bitvec_to_nat, Nat.shiftRight_eq_div_pow, Nat.mod_eq_of_lt this]
 
+theorem toInt_twoPow {w i : Nat} :
+    (BitVec.twoPow w i).toInt = if w ≤ i then (0 : Int) else (if i + 1 = w then -(1 <<< i:Int) else (1 <<< i)) := by
```

Comment:
There are lots of ways to phrase the RHS of the theorem, and I don't know which one is canonical.
Here's at least one more reasonable phrasing, where we only compute the sign in the `if-then-else`:

```
if w ≤ i then (0 : Int) 
else (if i + 1 = w then -1 else 1) * (1 <<< i) 
```

----

And here's a more perverse one which uses the linear function that sends `true` to `-1` and `false` to `1`, which lets us write the boolean `0/1` into an arithmetic `+1/-1` . This is not a pattern we use, but arose in my mind as a pattern that may help write simpler statements:

```
if w ≤ i then (0 : Int) 
else (1 - 2 * (Bool.ofInt (decide (i + 1 = w)))) * (1 <<< i)
```